app.js

```diff
@@ -29,23 +35,32 @@ const createDeleteButton = (todo) => {
 
   return deleteButton;
 };
-const handleEdit = (inputElement, editButton) => {
-  if (inputElement.getAttribute("data-state") === "edit") {
-    editButton.innerText = "Update";
-    inputElement.removeAttribute("readonly");
-    inputElement.setAttribute("data-state", "update");
-    return;
-  }
+const handleEdit = (inputElement, editButton, todo, checkbox) => {
+  if (inputElement.classList.contains("completed")) return;
```

Comment:
Checking `classList`, `innerText`, `innerHTML` can be flaky and could easily break as these things are subject to change. Let's utilise the data attribute here

```suggestion
  if (inputElement.getAttribute('state') === "completed"){
      return;
  }
```